test/integration/modules/edgecore/eventbus/eventbus_test.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"database/sql"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/kubeedge/kubeedge/test/integration/utils/common"
+
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	_ "github.com/mattn/go-sqlite3"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+//DeviceUpdate device update
+type DeviceUpdate struct {
+	State string `json:"state,omitempty"`
+}
+
+//the struct of device
+type Device struct {
+	id          []uint8
+	name        string
+	description string
+	state       string
+	lastOnline  []uint8
+}
+
+//Function to access the edge_core DB and return the device state.
+func getDeviceStateFromDB(deviceId string) string {
+	var device Device
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		common.Failf("Failed to get PWD: %v", err)
+		os.Exit(1)
+	}
+	destpath := filepath.Join(pwd, "../../edge.db")
+	db, err := sql.Open("sqlite3", destpath)
+	if err != nil {
+		common.Failf("Open Sqlite DB failed :%v", err)
+	}
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM device", 0)
+	if err != nil {
+		common.Failf("Query Sqlite DB failed: %v", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		err = rows.Scan(&device.id, &device.name, &device.description, &device.state, &device.lastOnline)
+		if err != nil {
+			common.Failf("Failed to scan DB rows: %v", err)
+		}
+
+		if string(device.id) == deviceId {
+			break
+		}
+	}
+	return device.state
+}
+
+//Run Test cases
+var _ = Describe("Event BUS Testing", func() {
+	var Token_client Token
+	var ClientOpts *MQTT.ClientOptions
+	var Client MQTT.Client
+	Context("Publish on eventbus topics through MQTT external broker", func() {
```

Comment:
in the config.json , the mqtt server url has port 1884 , this is the internal broker.
are we running the integration tests on the internal mqtt broker ? 

see https://github.com/kubeedge/kubeedge/pull/75/files for implementation of internal mqtt broker.

Reply:
@rohitsardesai83, Yes, we are running integration test on internal broker as current kubeedge code is supporting internal broker mode by default which is on 1884, the exter broker kubeedge will not start unless it is configured.

Going forward i will add few more test fot testing on external broker by changing the mode from internal to external( this requires changing the mode in edge.yaml and restart the edge_core process, this requires some more scripting need to be added to handle this scenario.).

But currenlty integration suite will run on internal port 1884.

Reply:
in that case , the description here needs to be updated external -> internal.

Reply:
Yes, updated.